core/src/sumeragi/main_loop.rs

```diff
@@ -241,11 +242,32 @@ impl Sumeragi {
         assert_eq!(self.wsv.height(), 0);
         assert_eq!(self.wsv.latest_block_hash(), None);
 
-        let transactions: Vec<_> = genesis_network
-            .transactions
-            .into_iter()
-            .map(AcceptedTransaction::accept_genesis)
-            .collect();
+        let transactions: Vec<_> = {
+            let mut transactions: Vec<_> = genesis_network.transactions.into_iter().collect();
+
+            // Copy sumeragi config params into chain.
+            let parameters = ParametersBuilder::new()
+                .add_parameter(BLOCK_TIME, self.block_time.as_millis())
+                .expect("should not fail")
+                .add_parameter(COMMIT_TIME_LIMIT, self.commit_time.as_millis())
+                .expect("should not fail")
+                .add_parameter(MAX_TRANSACTIONS_IN_BLOCK, self.max_txs_in_block as u32)
+                .expect("should not fail")
+                .into_set_parameters();
+
+            transactions.push(GenesisTransaction(
+                GenesisTransactionBuilder {
+                    isi: vec![InstructionExpr::from(parameters)],
+                }
+                .sign(genesis_network.genesis_key_pair)
+                .expect("Must succeed"),
+            ));
+
+            transactions
+                .into_iter()
+                .map(AcceptedTransaction::accept_genesis)
+                .collect()
+        };
```

Comment:
```suggestion
        let transactions: Vec<_> = {
            // Copy sumeragi config params into chain.
            genesis_network.set_parameters(ParametersBuilder::new()
                .add_parameter(BLOCK_TIME, self.block_time.as_millis())
                .expect("should not fail")
                .add_parameter(COMMIT_TIME_LIMIT, self.commit_time.as_millis())
                .expect("should not fail")
                .add_parameter(MAX_TRANSACTIONS_IN_BLOCK, self.max_txs_in_block as u32)
                .expect("should not fail")
                .into_set_parameters()
            );
```

this is how you should do it, and everything else should be in the genesis crate

Reply:
In that case they should not exist then? I think from a UX perspective it's way worse for a network operator to modify genesis.json. I will remove this but it does introduce a bug. Iroha users changing the block size in the config then being upset when nothing is different.

Reply:
> In that case they should not exist then? 

Yes, afaik @mversic and @0x009922 has ongoing discussion to preserve on chain parameter only in `genesis`.